packages/dds/tree/docs/wip/schema-evolution.md

```diff
@@ -0,0 +1,583 @@
+# SharedTree Schema Evolution
+
+This document provides the necessary background for understanding SharedTree's strategy for conforming data to a given user-provided schema.
```

Comment:
An interesting question to ask here: should title / this intro paragraph be focused solely on schema, or should it be inclusive and include data migration / transformation? The former includes the latter, but not necessarily the other way around. The example I used below - what if I want to add some prefix to the data in some field? It stays of string type.